### PR TITLE
MSSQL: Add MSSQLOptions

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3171.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [x] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/agent_plugins/exploiters/mssql/src/mssql_options.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class MSSQLOptions(InfectionMonkeyBaseModel):
+    target_ports: List[NetworkPort] = Field(
+        default=[1433],
+        description="A list of TCP ports that the MSSQL exploiter will try to exploit.",
+    )
+    try_all_discovered_tcp_ports: bool = Field(
+        default=False,
+        description="Attempt to exploit MSSQL on all TCP ports discovered from network scanning.",
+    )
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/mssql/src/mssql_options.py:17]: I don't know that we want to try on all discovered ports. Just ports where we have positively identified that MSSQL is running.

File: monkey/agent_plugins/exploiters/mssql/src/mssql_options.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        " MSSQL server to download the agent binary.",
+    )
+    login_timeout: float = Field(
+        gt=0.0,
+        default=15.0,
+        description="The maximum time (in seconds) to wait for the SQL Server login to complete",
+    )
+    query_buffer_time: float = Field(
+        gt=0.0, default=0.5, description="The time (in seconds) to wait between SQL queries."
+    )
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/mssql/src/mssql_options.py:36]: We can probably drop this.

File: monkey/agent_plugins/exploiters/mssql/src/mssql_options.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        gt=0.0,
+        default=60.0,
+        description="The maximum time (in seconds) to wait for a successfully exploited"
+        " MSSQL server to download the agent binary.",
+    )
+    login_timeout: float = Field(
+        gt=0.0,
+        default=15.0,
+        description="The maximum time (in seconds) to wait for the SQL Server login to complete",
+    )
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/mssql/src/mssql_options.py:33]: Is there any known reason to distinguish this from `query_timeout`? Perhaps we could have just one "timeout" that is the amount of time to wait for a response from the server. Whether it's a login response or a query response probably doesn't matter.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/agent_plugins/exploiters/mssql/src/mssql_options.py (hunk 3 above)
Comment (ilija-lazoroski) [reply monkey/agent_plugins/exploiters/mssql/src/mssql_options.py:33]: I thought that make sense. In either case both of them raise the same error if we pass the timeout so I don't think it really matters. I will merge them and call its `server_timeout`

File: monkey/agent_plugins/exploiters/mssql/src/mssql_options.py (hunk 1 above)
Comment (ilija-lazoroski) [reply monkey/agent_plugins/exploiters/mssql/src/mssql_options.py:17]: `try_all_mssql_open_ports` ?

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**